docs/articles/documentation/test-api/actions/take-screenshot.md

```diff
@@ -6,23 +6,99 @@ checked: true
+`path`&#160;*(optional)* | String | The relative path and the name of the screenshot file to be created. Resolved from the *screenshot directory* specified by using the [runner.screenshots](../../using-testcafe/programming-interface/runner.md#screenshots) API method or the [screenshots](../../using-testcafe/command-line-interface.md#-s-path---screenshots-path) command line option.
+`options`&#160;*(optional)*   | Object | Options that define how the screenshot will be taken. See details below.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Take a screenshot of a fieldset', async t => {
+    await t
+        .click('#reusing-js-code')
+        .click('#continuous-integration-embedding')
+        .takeElementScreenshot(Selector('fieldset').nth(1), 'my-fixture/important-features.png');
+});
+```
+
+By default, paths to which screenshots are saved are specified as:
+
+* `{currentDate}\test-{testIndex}\{userAgent}\{screenshotIndex}.png` if the [quarantine mode](../../using-testcafe/command-line-interface.md#-q---quarantine-mode) is disabled;
+* `{currentDate}\test-{testIndex}\run-{quarantineAttempt}\{userAgent}\{screenshotIndex}.png` if the [quarantine mode](../../using-testcafe/command-line-interface.md#-q---quarantine-mode) is enabled.
+
+The `options` object contains the following properties.
+
+Property        | Type | Description
+--------------- | ---- | ------------
+`scrollTargetX`,&#160;`scrollTargetY` | Number | If the target element is too big to fit into the browser window, the page will be scrolled to put this point to the center of the viewport. The coordinates of this point are calculated relative to the target element. If the numbers are positive, the point is positioned relative to the top-left corner of the element. If the numbers are negative, the point is positioned relative to the bottom-right corner. If the target element fits into the browser window, these properties have no effect.
+`includeMargins` | Boolean | Set this property to `true` to include target element's margins in the screenshot.
+`crop`           | Object  | Allows you to crop the target element on the screenshot.
+
+An object assigned to the `crop` property has the following fields.
+
+Field                                        | Type   | Description
+-------------------------------------------- | ------ | --------------
+`width`,&#160;`height`                       | Number | The width and height of the cropping rectangle.
+`left`,&#160;`right`, `bottom`,&#160;`top`     | Number | The position of the cropping rectangle.
+
+This is how the crop rectangle position and dimensions are determined.
+
+* If the `left`, `right`, `bottom` and `top` fields are specified, the `width` and `height` fields are ignored.
```

Comment:
Why? We need only coordinates. It's not difficult at all.
```js
bottom = top + height;
right = left + width;
```
At my opinion this option set looks confusing.

Reply:
If you want to crop borders, it's more convenient to write `{left: 2, top: 2, bottom: -2, right: -2}` than getting `width` and `height` and calculating `right`/`left` in the test code.

Reply:
So let's just keep `{ left, top, bottom, right }` in options. I mean we don't need width and height.

Reply:
You can use these options for cropping image of big element.  It's quite conveniently e.g. to use bottom border (`right` coord) and just set width/height area which you interest in. May be should try to describe them clearer.

Reply:
It's actually pretty simple and idiomatic. 

You can adjust cropping in vertical and horizontal directional. In each direction you can specify cropping using two points (`left` + `right`, `top`+`bottom`) or using  a point and length (`left` + `width`, `right` + `width`,  `top` + `height`, `bottom` + `height`). 

In my point of view, i's idomatic, because these rules are very similar to CSS rules.

Reply:
Maybe it will be more clear if we describe properties in this way:
```
bottom - he bottom margin edge for a cropping region
left - the left margin edge for a cropping region
right - the right margin edge for a cropping region
top - the top margin edge for a cropping region
width - the width of a cropping region
height - the width of a cropping region

If both margins in vertical or horizontal  direction are spcified, width or height will be ignored.
```
https://www.w3schools.com/css/css_positioning.asp

Reply:
if this scheme really matches the way CSS works than I'm for it

Reply:
After some discussion, it's clear that negative coordinates cover all the cases where `width` and `height` would be useful, so they mustn't be included in the `crop` option.

Reply:
So, regarding negative values, does it work this way now?

Field    | Type   | Description
-------- | ------ | --------------
`top`    | Number | The top edge of the cropping rectangle. If a negative number is passed, the top edge coordinate is calculated from the element's bottom edge.
`left`   | Number | The left edge of the cropping rectangle. If a negative number is passed, the left edge coordinate is calculated from the element's right edge.
`bottom` | Number | The bottom edge of the cropping rectangle. If a negative number is passed, the bottom edge coordinate is calculated from the element's top edge.
`right`  | Number | The right edge of the cropping rectangle. If a negative number is passed, the right edge coordinate is calculated from the element's left edge.